adrs/0153-checkout-v2.md

```diff
@@ -40,9 +40,12 @@ We want to take this opportunity to make behavioral changes, from v1. This docum
     default: ${{ github.token }}
   ssh-key:
```

Comment:
Many actions take a dependency on auth being configured. So for the workflow repo, even after the initial fetch we need to leave the remote URL configured as HTTPS.

We could always add a separate input `prefer-ssh` to let users opt in to fetching the main repo using SSH. I worry a little bit about relative URLs.

Do we need to worry about LFS? I don't it affects this decision - using the token should be fine?

Reply:
If we are adding extra inputs for this scenario we may want to consider having the inputs be opt in for users who need this functionality. It would be less entangled in the mainline scenario.

Its a bit verbose but we could do:
`submodules-ssh-key`
`submodules-token` 
They would default to `ssh-key` or  `token`.

This would enable us to work on any repo (not just the repo that kicked off the action). Users who don't need this scenario would ignore these inputs (and the consequences from them)

Reply:
> Many actions take a dependency on auth being configured. So for the workflow repo, even after the initial fetch we need to leave the remote URL configured as HTTPS.

Can we add this information to `persist-credentials`

Reply:
I think we're saying different things. This proposal changes the behavior for the main repo only ("the repository the workflow is running in").

Reply:
Also i think it's ok to still configure the ssh command. Remotes that use HTTPS wont use it.

Reply:
Discussed offline. The separate inputs `submodules-ssh-key` and `submodules-token` makes more sense.

Reasons customers would want to use ssh key for the workflow repo:
- Trigger additional run on push
- Push a commit that edits the workflow file